agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/InheritedAttributesSpanProcessor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.microsoft.applicationinsights.agent.internal.configuration.Configuration;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InheritedAttributesSpanProcessor implements SpanProcessor {
+
+  private final List<AttributeKey<?>> inheritAttributeKeys;
+
+  public InheritedAttributesSpanProcessor(
+      List<Configuration.InheritedAttribute> inheritedAttributes) {
+    this.inheritAttributeKeys =
+        inheritedAttributes.stream()
+            .map(Configuration.InheritedAttribute::getAttributeKey)
+            .collect(Collectors.toList());
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void onStart(Context parentContext, ReadWriteSpan span) {
+    Span parentSpan = Span.fromContextOrNull(parentContext);
+    if (parentSpan == null) {
+      return;
+    }
+    if (!(parentSpan instanceof ReadableSpan)) {
+      return;
+    }
+    ReadableSpan parentReadableSpan = (ReadableSpan) parentSpan;
+
+    for (AttributeKey<?> inheritAttributeKey : inheritAttributeKeys) {
+      Object value = TempGetAttribute.getAttribute(parentReadableSpan, inheritAttributeKey);
+      if (value != null) {
+        span.setAttribute((AttributeKey<Object>) inheritAttributeKey, value);
```

Comment:
are inheritAttributes part of the customDimensions in the payload?

Reply:
yes